src/snowflake/snowpark/modin/plugin/extensions/timedelta_index.py

```diff
@@ -407,17 +412,39 @@ def mean(
 
         Returns
         -------
-            scalar Timestamp
+            scalar Timedelta
+
+        Examples
+        --------
+        >>> idx = pd.to_timedelta([1, 2, 3, 1], unit='D')
+        >>> idx
+        TimedeltaIndex(['1 days', '2 days', '3 days', '1 days'], dtype='timedelta64[ns]', freq=None)
+        >>> idx.mean()
+        Timedelta('1 days 18:00:00')
 
         See Also
         --------
         numpy.ndarray.mean : Returns the average of array elements along a given axis.
         Series.mean : Return the mean value in a Series.
-
-        Notes
-        -----
-        mean is only defined for Datetime and Timedelta dtypes, not for Period.
         """
+        if axis:
+            # Native pandas raises IndexError: tuple index out of range
+            # We raise a different more user-friendly error message.
+            raise ValueError(
+                f"axis should be 0 for TimedeltaIndex.mean, found '{axis}'"
```

Comment:
This is a bit weird. If `axis` has to be 0, what's the point of having it as a parameter in the first place?

Reply:
yeah, pandas is 30% weird, 70% useful.